src/cca/model/vpn_device.ts

```diff
@@ -0,0 +1,9 @@
+// Represents a VPN device that can redirect traffic to a Socks server.
+export interface VpnDevice {
+  // Starts the VPN, redirecting the traffic to a Socks proxy running on the given port.
+  // You can pass a callback to be notified when it gets disconnected.
+  start(port: number, onDisconnect: (msg: string) => void) : Promise<string>;
```

Comment:
We can't, because we always force localhost. I'd pass an `Endpoint` if we supported any IP.